packages/app/src/components/DBTraceWaterfallChart.tsx

```diff
@@ -22,233 +27,338 @@ type SpanRow = {
+  }[] = [];
+
+  if (source.kind === SourceKind.Trace) {
+    selectOption = [
+      {
+        valueExpression: alias.Body,
+        alias: 'Body',
+      },
+      {
+        valueExpression: alias.Timestamp,
+        alias: 'Timestamp',
+      },
+      {
+        // in Seconds, f64 holds ns precision for durations up to ~3 months
+        valueExpression: alias.Duration,
+        alias: 'Duration',
+      },
+      {
+        valueExpression: alias.SpanId,
+        alias: 'SpanId',
+      },
+      {
+        valueExpression: alias.ParentSpanId,
+        alias: 'ParentSpanId',
+      },
+      ...(alias.StatusCode
+        ? [
+            {
+              valueExpression: alias.StatusCode,
+              alias: 'StatusCode',
+            },
+          ]
+        : []),
+      ...(alias.ServiceName
+        ? [
+            {
+              valueExpression: alias.ServiceName,
+              alias: 'ServiceName',
+            },
+          ]
+        : []),
+    ];
+  } else if (source.kind === SourceKind.Log) {
+    selectOption = [
+      {
+        valueExpression: alias.Body,
+        alias: 'Body',
+      },
+      {
+        valueExpression: alias.Timestamp,
+        alias: 'Timestamp',
+      },
+      {
+        valueExpression: alias.SpanId,
+        alias: 'SpanId',
+      },
+      ...(alias.SeverityText
+        ? [
+            {
+              valueExpression: alias.SeverityText,
+              alias: 'SeverityText',
+            },
+          ]
+        : []),
+      ...(alias.ServiceName
+        ? [
+            {
+              valueExpression: alias.ServiceName,
+              alias: 'ServiceName',
+            },
+          ]
+        : []),
+    ];
+  }
+  const config = {
+    select: selectOption,
+    from: source.from,
+    timestampValueExpression: alias.Timestamp,
+    where: `${alias.TraceId} = '${traceId}'`,
+    limit: { limit: 10000 },
+    connection: source.connection,
+  };
+  return { config, alias, type: source.kind };
+}
+
+function useFetchingData({
+  config,
+  dateRangeStartInclusive,
+  dateRange,
+}: {
+  config: {
+    select: {
+      valueExpression: string;
+      alias: string;
+    }[];
+    from: {
+      databaseName: string;
+      tableName: string;
+    };
+    timestampValueExpression: string;
+    where: string;
+    limit: {
+      limit: number;
+    };
+    connection: string;
+  };
+  dateRangeStartInclusive: boolean;
+  dateRange: [Date, Date];
+}) {
+  const query: ChartConfigWithDateRange = useMemo(() => {
+    return {
+      ...config,
+      dateRange,
+      dateRangeStartInclusive,
+      orderBy: [
+        {
+          valueExpression: config.timestampValueExpression,
+          ordering: 'ASC',
+        },
+      ],
+    };
+  }, [config, dateRange]);
+  return useOffsetPaginatedQuery(query);
+}
+
 function useSpansAroundFocus({
-  traceTableModel,
+  tableModel,
   focusDate,
   dateRange,
   traceId,
 }: {
-  traceTableModel: TSource;
+  tableModel: TSource;
```

Comment:
this is some inconsistency in our app but I think we're trying to use `source` instead of `model` now. so `tableSource`

Reply:
sure, I will use source instead of table.